addons/mail/models/ir_actions_server.py

```diff
@@ -90,6 +116,30 @@ def _compute_available_model_ids(self):
                 action.available_model_ids = mail_models.ids
         super(IrActionsServer, self - mail_thread_based)._compute_available_model_ids()
 
+    @api.depends('state')
+    def _compute_discuss_channel_ids(self):
+        to_reset = self.filtered(lambda act: act.state != 'discuss_post')
```

Comment:
I'm not sure it works that way: for some records there won't be a computed value.
This feature, I think, should be an inverse on the state field, or encapsulated within an onchange
This comment is valid for the other compute methods that aim to do the same things